src/components/schema-editor/schema-editor.component.tsx

```diff
@@ -2,49 +2,132 @@ import React from 'react';
-          <p className={styles.heading}>{t('schemaError', "There's an error in your schema.")}</p>
-          <p>{invalidJsonErrorMessage}</p>
-        </div>
-      ) : null}
+  const validateJSON = (content: string, schema) => {
+    const ajv = new Ajv({ allErrors: true, jsPropertySyntax: true });
+    const validate = ajv.compile(schema);
+    const parsedContent = JSON.parse(content);
+    const isValid = validate(parsedContent);
+
+    if (!isValid) {
+      const jsonLines = content.split('\n');
+      const errorMarkers = validate.errors.map((error) => {
+        let lineNumber = -1;
+        const instancePath = error.instancePath.replace(/^\./, '');
+        for (let i = 0; i < jsonLines.length; i++) {
+          if (jsonLines[i].includes(instancePath)) {
+            lineNumber = i;
+            break;
+          }
+        }
+        const message =
+          error.keyword === 'type'
+            ? `Invalid Type: ${error.instancePath.substring(1)} ${error.message}`
+            : `${capitalize(Object.keys(error.params)[0])}: ${
+                error.message.charAt(0).toUpperCase() + error.message.slice(1)
+              }`;
+
+        return {
+          startRow: lineNumber,
+          startCol: 0,
+          endRow: lineNumber,
+          endCol: 1,
+          className: 'error',
+          text: message,
+          type: 'error',
+        };
+      });
+      setErrors(errorMarkers);
+    } else {
+      setErrors([]);
+    }
+  };
 
+  const handleChange = (newValue: string) => {
+    onSchemaChange(newValue);
+    fetchAndParseSchema()
+      .then((schema) => {
+        validateJSON(newValue, schema);
+      })
+      .catch((error) => {
+        console.error('Error fetching/parsing schema:', error);
+      });
+  };
+
+  const ErrorMessages = () => (
+    <Toggletip className={styles.toggletipContainer} align="top-left">
+      <ToggletipButton label="Show information">
+        <WarningAlt />
+      </ToggletipButton>
+      <ToggletipContent>
+        {errors.map((e) => (
+          <div>{e.text}</div>
+        ))}
+        <ToggletipActions>
+          <Link href="https://json.openmrs.org/form.schema.json" target="_blank">
+            Reference Schema
+          </Link>
+        </ToggletipActions>
+      </ToggletipContent>
+    </Toggletip>
+  );
+
+  return (
+    <div>
+      {errors.length ? <ErrorMessages /> : null}
       <AceEditor
         style={{ height: '100vh', width: '100%' }}
         mode="json"
         theme="textmate"
         name="schemaEditor"
-        onChange={onSchemaChange}
+        onChange={handleChange}
         fontSize={15}
         showPrintMargin={false}
         showGutter={true}
         highlightActiveLine={true}
         value={stringifiedSchema}
         setOptions={{
-          enableBasicAutocompletion: false,
-          enableLiveAutocompletion: false,
+          enableBasicAutocompletion: true,
+          enableLiveAutocompletion: true,
           displayIndentGuides: true,
```

Comment:
I remember us testing the `enableBasicAutocompletion`, did you manage to get this working?